### PR TITLE
Add barracuda/0.1.0 integration (snapshot)

### PR DESCRIPTION
**Pull request by adriansr**
This adds the first rsa2elk integration package for Barracuda WAF devices.

After this one, I'll add the next 19 rsa2elk devices in a single PR if possible. They all follow the exact same structure than this one.

## Screenshots

Screenshot 2020-07-23 at 15 41 58

Screenshot 2020-07-23 at 15 42 38

With logo:

Screenshot 2020-07-28 at 12 02 13

### CODE REVIEW

File: packages/barracuda/0.1.0/dataset/waf/agent/stream/stream.yml.hbs [hunk 1]
```diff
@@ -0,0 +1,3721 @@
...
+- script:
+    lang: javascript
+    params:
+      ecs: true
+      rsa: {{rsa_fields}}
+      tz_offset: {{tz_offset}}
+      keep_raw: {{keep_raw_fields}}
+      debug: {{debug}}
+    source: |
+      // Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```
Comment (mtojek) [inline packages/barracuda/0.1.0/dataset/waf/agent/stream/stream.yml.hbs:26]: @nchaulet @michalpristas Could you please confirm that such huge blobs should be supported by all parties?

File: packages/barracuda/0.1.0/dataset/waf/manifest.yml [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  - name: tcp_host
+    type: text
+    title: TCP host to listen on
+    multi: false
+    required: true
+    show_user: true
+    default: localhost
+  - name: tcp_port
+    type: integer
+    title: TCP Port to listen on
```
Comment (mtojek) [inline packages/barracuda/0.1.0/dataset/waf/manifest.yml:81]: nit: port

File: packages/barracuda/0.1.0/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        title: Collect logs from Barracuda Web Application Firewall via UDP
+        description: Collecting syslog from Barracuda Web Application Firewall via UDP
+      - type: tcp
+        title: Collect logs from Barracuda Web Application Firewall via TCP
+        description: Collecting syslog from Barracuda Web Application Firewall via TCP
+      - type: file
+        title: Collect logs from Barracuda Web Application Firewall via file
+        description: Collecting syslog from Barracuda Web Application Firewall via file.
+# No icon
+icon:
```
Comment (andrewstucki) [inline packages/barracuda/0.1.0/manifest.yml:31]: I'm pretty sure we can easily find a barracuda svg to dump on here, is there a reason not to add an icon?

File: packages/barracuda/0.1.0/docs/README.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Barracuda integration
+
+This integration is for Barracuda device's logs. It includes the following
+datasets for receiving logs over syslog or read from a file:
+- `waf` dataset: supports Barracuda Web Application Firewall logs.
+
+### Waf
+
+The `waf` dataset collects Barracuda Web Application Firewall logs.
```
Comment (andrewstucki) [inline packages/barracuda/0.1.0/docs/README.md:9]: So, generally the way this file gets generated in the integrations directory, it contains a listing of all of the fields along with their documentation. Does this need to be consistent here too?

File: packages/barracuda/0.1.0/dataset/waf/agent/stream/udp.yml.hbs [hunk 5]
```diff
@@ -0,0 +1,3716 @@
+udp:
+host: "{{udp_host}}:{{udp_port}}"
+tags: {{tags}}
```
Comment (andrewstucki) [inline packages/barracuda/0.1.0/dataset/waf/agent/stream/udp.yml.hbs:3]: See [comment](https://github.com/elastic/integrations/pull/186#discussion_r458441240) I made on the Suricata package, but in filebeat we pipe this to `toJSON`, so I'm assuming we need it rendered as a yaml array. The way this is right now it'll output as a comma-separated string.

File: packages/barracuda/0.1.0/manifest.yml (hunk 3 above)
Comment (adriansr) [reply packages/barracuda/0.1.0/manifest.yml:31]: Afaik we don't have (yet) clearance from Barracuda for the use of their logo in a commercial product. I see a lot of logos in other packages, @mtojek is someone from the ingest manager team handling this? /cc @jamiehynds

Comment (mtojek) [reply packages/barracuda/0.1.0/manifest.yml:31]: Nobody. Currently used logos are imported from existing open-source repos of Elastic. In general in terms of the ingest manager, the point of contact is @ruflin .

Comment (andrewstucki) [reply packages/barracuda/0.1.0/manifest.yml:31]: So, I pulled the logo from https://www.elastic.co/integrations for my WIP integration. Even though I once went to law school, IANAL, so not entirely sure what additional consent we need from those other companies for inclusion of their logos here.

Comment (adriansr) [reply packages/barracuda/0.1.0/manifest.yml:31]: According to @jamiehynds we're OK adding the logos. Will update the PR.

**Comment by adriansr** [comment]
- Changed categorization to Network + Security
- Added logo:

Screenshot 2020-07-28 at 12 02 13

**Pull request merged**